### DISCUSSION TITLE
Help adding options to formula

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#5385] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: e0bc557e7b991cb23583679e1cf1c8a92b793aeb
Last commit: 24 hours ago
Core tap JSON: 21 May 03:34 UTC
Core cask tap JSON: 21 May 03:34 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.1 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.1/bin/ruby
CPU: octa-core 64-bit arm_ibiza
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The formula in question is `chibi-scheme`, which as of yet has no options so I'd have to add the whole option infrastructure from scratch, which I don't know how to do. It exposes a litany of compile options, but the only benefit of most of them is to reduce the binary size, which I don't think Homebrew users are all that worried about; however, there are some that impact semantics and performance, which could be a concern. The ones in particular that appeal to me are:
``` c
/* Listed here as they appear in <chibi/features.h>,
 * commented out, with their _non_-default values */

/* Enables a situationally non-compliant binding scheme
 * that smoothens writing shared macros */
#define SEXP_USE_STRICT_TOPLEVEL_BINDINGS 0
/* Disables string mutation, permitting optimisations */
#define SEXP_USE_MUTABLE_STRINGS 0
/* Precomputes string offsets, curbing linear-time indexing */
#define SEXP_USE_STRING_INDEX_TABLE 1
/* Self-explanatory */
#define SEXP_USE_ALIGNED_BYTECODE
```

**Comment by SMillerDev** [comment]
> Note: [option](https://rubydoc.brew.sh/Formula#option-class_method)s are not allowed in Homebrew/homebrew-core as they are not tested by CI.

https://docs.brew.sh/Formula-Cookbook#adding-optional-steps

**Comment by ashinn** [comment]
FWIW, the defaults for `SEXP_USE_STRICT_TOPLEVEL_BINDINGS` and `SEXP_USE_MUTABLE_STRINGS` are not going to change since they're required for R7RS compatibility.  I'm not a fan of `SEXP_USE_STRING_INDEX_TABLE` but I appear to be the only person on this planet who holds that opinion.  `SEXP_USE_ALIGNED_BYTECODE` should really be inferred automatically from the architecture.

**Comment by ghost** [comment]
Of course, I’m not talking about changing the _defaults_ — I'm talking about adding package options to this specific formula. Personally, the included modules will be plenty for my purposes, and I have no desire to run my code on another interpreter; but I agree, the defaults should minimise headaches.

I mentioned getting you on board because, if options aren’t allowed where the package is now, then an alternate package with options will need to be somewhere else; no way was I going to ask you to move it into a tap, but moving it somewhere that didn’t need to be tapped might have been on the table.

(Also re. aligned bytecode, I’m indifferent to the default but I’ll explain my affinity for the option: from my understanding, even if an architecture can perform unaligned accesses, such accesses are typically slower and less predictable than aligned. I know the OCaml platform does not emit them for this reason.)

[the remark this answers is not in the document]
**Reply by SMillerDev** [reply]
Yeah, you'll have to make your own tap since the options ban is not based on upstream preferences but on keeping Homebrew default indexes maintainable.
